sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollectionExpressionsSuite.scala

```diff
@@ -934,12 +1071,25 @@ class CollectionExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper
         EmptyRow,
         s"sequence boundaries: 0 to 2678400000000 by -${28 * MICROS_PER_DAY}")
 
+      checkExceptionInExpression[IllegalArgumentException](
+        new Sequence(
+          Literal(Date.valueOf("1970-01-01")),
+          Literal(Date.valueOf("1970-02-01")),
+          Literal(Period.ofMonths(-1))),
+        EmptyRow,
+        s"sequence boundaries: 0 to 2678400000000 by -${28 * MICROS_PER_DAY}")
```

Comment:
hmm, 28 because we assume 28 days per month there https://github.com/apache/spark/blob/9af338cd685bce26abbc2dd4d077bde5068157b1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala#L2742 ?

By introducing new interval types, we tried to avoid such assumption.

Reply:
Good found.
It seems we should change the behavior of `CalenderInterval` so as it could avoid such assumption.
Or the assumption looks like is a bug for the `CalenderInterval`.
If so , could we fix the bug in another PR?

Reply:
In further, microsPerDay just used to estimated length of the sequences. We just need to improve the exception message so as avoid output the assume value.

Reply:
ok. I see.